sierra2mlir/src/types/mod.rs

```diff
@@ -93,6 +93,76 @@ impl<'ctx> Compiler<'ctx> {
                         SierraType::Struct { ty: struct_type, field_types: types },
                     );
                 }
+                "Enum" => {
+                    // https://mapping-high-level-constructs-to-llvm-ir.readthedocs.io/en/latest/basic-constructs/unions.html#tagged-unions
+
+                    // in case the enum has variants with the same type, just store a uniqued one.
+                    let mut unique_enum_types = HashSet::new();
+                    // all the enum variant types, even repeated, for later use when storing the variant value.
+                    let mut enum_variant_types = vec![];
+
+                    let _user_type = match &type_decl.long_id.generic_args[0] {
+                        GenericArg::UserType(x) => x,
+                        _ => {
+                            unreachable!("first arg on struct libfunc should always be a user type")
```

Comment:
I'd use `todo!()` if we're going to add an error type, or just `panic!()` directly because `unreachable!()` implies that the code shouldn't be reachable, but we can't really be sure about that since the sierra code is provided externally.

Reply:
i think unreachable is fine here, since this is the safe version it is essentially a panic! wrapper with a semantic